<a href="https://colab.research.google.com/github/AkshatkMalik/python-data-science-toolkit/blob/main/hospitality_hypothesis_test.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

**ASSIGNENT-4**

**Hypothesis Testing Report: Bombay Hospitality Ltd. Operating Cost Analysis**

**Background/Objective**

Bombay Hospitality Ltd. uses the franchise method of production for making exotic Norwegian meals in New England. The cost of operation for a single franchise in a week ($W$) can be expressed using a linear formula as follows:

**$W = \$1,000 + \$5X$**

where $X$ denotes the number of units produced in a week. The restaurant owners have lately stated that the above formula is not applicable to calculate the cost of operations since it appears that the weekly cost of operation is higher than the one mentioned above.

**Statement of Hypotheses**

In order to check if the operating costs have gone up, we perform a one-sided test:

Hypothesis: $H_0$ = the true average weekly operating cost is equal to the theoretical model cost value ($\mu = \$4,000$). (Model is valid).

Alternative Hypothesis: $H_a$ = the true average weekly operating cost is larger than the theoretical model cost value ($\mu > \$4,000$). (Owners of the restaurant are right; costs have gone up).

**Calculations of Parameters and Methods**

Theoretical Mean ($\mu$): Estimated from the equation $W = 1,000 + 5X$, given $X = 600$ units:

$$\mu = 1000 + 5(600) = \$4,000$$

Standard Deviation of Cost ($\sigma_W$): Given that $W = 1000 + 5X$ and the standard deviation for the number of units $X$ is $25$:$$\sigma_W = 5 \times 25 = \$125$$

Sample Mean ($\bar{x}$): Rs. $3,050$. (It should be noted that the sample mean used in the problem is based on the actual sample).

**Python Implementation Code**

In [1]:
import math
import scipy.stats as stats

# given parameters

units_produced = 600
base_cost = 1000
cost_slope = 5

# Theoretical Mean calculation

mu_theory = base_cost + (cost_slope * units_produced)  # $4,000

# Sample data parameters

sample_mean = 3050       # Sample mean weekly cost
sigma_x = 25             # Standard deviation of units produced X
n = 25                   # Sample size

# Calculate standard deviation of W (sigma)

sigma_w = cost_slope * sigma_x  # 5 * 25 = 125

# 2. Calculate the Test Statistic (Z)

# Formula: Z = (x_bar - mu) / (sigma / sqrt(n))
standard_error = sigma_w / math.sqrt(n)
z_stat = (sample_mean - mu_theory) / standard_error

# 3. Determine Critical Value for a right-tailed test with alpha = 0.05
alpha = 0.05
critical_value = stats.norm.ppf(1 - alpha)

# 4. Print structured result

print(" HYPOTHESIS TESTING RESULTS ")
print(f"Theoretical Mean (mu): {mu_theory}")
print(f"Sample Mean (x-bar): {sample_mean}")
print(f"Standard Error: {standard_error:.4f}")
print(f"Test Statistic (Z): {z_stat:.4f}")
print(f"Critical Value (alpha = 0.05): {critical_value:.4f}")

# 5. Automated Decision

print("\n DECISION ")
if z_stat > critical_value:
    print("Result: Reject the Null Hypothesis (H0). Evidence supports the owners' claim.")
else:
    print("Result: Fail to Reject the Null Hypothesis (H0). No strong evidence that costs are higher.")

 HYPOTHESIS TESTING RESULTS 
Theoretical Mean (mu): 4000
Sample Mean (x-bar): 3050
Standard Error: 25.0000
Test Statistic (Z): -38.0000
Critical Value (alpha = 0.05): 1.6449

 DECISION 
Result: Fail to Reject the Null Hypothesis (H0). No strong evidence that costs are higher.


**Test Statistic & Critical Value Results**

Test Statistic ($Z$): $-38.0000$$$\text{Computed as: } Z = \frac{3050 - 4000}{\frac{125}{\sqrt{25}}} = \frac{-950}{25} = -38.0$$

Critical Value: $1.645$ (obtained from the standard normal $Z$ distribution table for a one-tailed test at $\alpha = 0.05$).

**Conclusion & Final Decision**

**Rule for Decision:**

In the case of a one-tailed upper hypothesis test at $\text{significance level} = \alpha = 5\%$, the critical threshold value obtained from the standard normal distribution is $Z_{\text{crit}} = 1.645$. The rejection region consists of all values above the critical threshold value, that is, $Z > 1.645$.Our calculated value of test statistic is $Z = -38.0000$. Since $-38.0000$ is far below the critical value of $1.645$, it is well within the non-rejection region and hence, we do not reject $H_0$.

**Final Conclusion & Managerial Implication**

:Statistically, there is no evidence to prove the hypothesis made by the owners of the restaurant about the rise in the weekly cost incurred. On the contrary, the average value cost obtained from the sample, that is, $\text{Rs. } 3,050$ is far below the theoretical value of expected cost ($\$4,000$). Thus, Bombay Hospitality Ltd. can be assured that the developed cost model is statistically correct ($W = \$1,000 + \$5X$).